jib-core/src/main/java/com/google/cloud/tools/jib/ncache/DefaultCacheStorageWriter.java

```diff
@@ -75,6 +82,15 @@ private static void moveIfDoesNotExist(Path source, Path destination) throws IOE
 
       } catch (FileAlreadyExistsException alsoIgnored) {
         // Same reasoning
+
+      } catch (DirectoryNotEmptyException ex) {
+        // The file system cannot rename directories, so we must resort to copying the directory.
+        Files.createDirectory(destination);
+        try (Stream<Path> sourceFiles = Files.list(source)) {
+          for (Path sourceFile : sourceFiles.collect(Collectors.toList())) {
+            Files.copy(sourceFile, destination.resolve(sourceFile.getFileName()));
+          }
+        }
```

Comment:
Try removing the tmp source?

Reply:
The tmp source would be removed via the `TemporaryDirectory` mechanism - do you think it'd be good to just do that here as well?

Reply:
This method is not aware of `TemporaryDirectory`, so it won't hurt to try to keep the promise of "move". But then, we will probably ignore failures from deleting the source fails, so I'm fine with leaving it there, considering also that this manual copy is somewhat a corner case.